docs/src/content/en/docs/auth/auth0.mdx

```diff
@@ -0,0 +1,311 @@
+---
+title: "MastraAuthAuth0 Class"
+description: "Documentation for the MastraAuthAuth0 class, which authenticates Mastra applications using Auth0 authentication."
+---
+
+import { Tabs, Tab } from "@/components/tabs";
+
+# MastraAuthAuth0 Class
+
+The `MastraAuthAuth0` class provides authentication for Mastra using Auth0. It verifies incoming requests using Auth0-issued JWT tokens and integrates with the Mastra server using the `experimental_auth` option.
+
+## Prerequisites
+
+This example uses Auth0 authentication. Make sure to:
+
+1. Create an Auth0 account at [auth0.com](https://auth0.com/)
+2. Set up an Application in your Auth0 Dashboard
+3. Configure an API in your Auth0 Dashboard with an identifier (audience)
+4. Configure your application's allowed callback URLs, web origins, and logout URLs
+
+```env filename=".env" copy
+AUTH0_DOMAIN=your-tenant.auth0.com
+AUTH0_AUDIENCE=your-api-identifier
+```
+
+> **Note:** You can find your domain in the Auth0 Dashboard under Applications > Settings. The audience is the identifier of your API configured in Auth0 Dashboard > APIs.
+
+> For detailed setup instructions, refer to the [Auth0 quickstarts](https://auth0.com/docs/quickstarts) for your specific platform.
+
+## Installation
+
+Before you can use the `MastraAuthAuth0` class you have to install the `@mastra/auth-auth0` package.
+
+```bash copy
+npm install @mastra/auth-auth0@latest
+```
+
+## Usage examples
+
+### Basic usage with environment variables
+
+If you set the required environment variables (`AUTH0_DOMAIN` and `AUTH0_AUDIENCE`), you can initialize `MastraAuthAuth0` without any constructor arguments. The class will automatically read these environment variables as configuration:
+
+```typescript {2,8} filename="src/mastra/index.ts" showLineNumbers copy
+import { Mastra } from "@mastra/core/mastra";
+import { MastraAuthAuth0 } from '@mastra/auth-auth0';
+
+export const mastra = new Mastra({
+  // ..
+  server: {
+    // Automatically uses AUTH0_DOMAIN and AUTH0_AUDIENCE env vars
+    experimental_auth: new MastraAuthAuth0(),
+  },
+});
+```
+
+### Custom configuration
+
+```typescript {2,7-10} filename="src/mastra/index.ts" showLineNumbers copy
+import { Mastra } from "@mastra/core/mastra";
+import { MastraAuthAuth0 } from '@mastra/auth-auth0';
+
+export const mastra = new Mastra({
+  // ..
+  server: {
+    experimental_auth: new MastraAuthAuth0({
+      domain: 'your-tenant.auth0.com',
+      audience: 'your-api-identifier'
+    }),
+  },
+});
+```
+
+## Configuration
+
+The `MastraAuthAuth0` class can be configured through constructor options or environment variables.
+
+### Environment Variables
+
+- `AUTH0_DOMAIN`: Your Auth0 domain (e.g., `your-tenant.auth0.com`)
```

Comment:
Is this duplicated, you have the env vars in the prerequsites section at the top.